app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt

```diff
@@ -175,70 +176,113 @@ abstract class GitOperation(fileDir: File, internal val callingActivity: Activit
+                            .setMessage(callingActivity.resources.getString(R.string.git_operation_unable_to_open_ssh_key_message))
+                            .setPositiveButton(callingActivity.resources.getString(R.string.dialog_ok)) { _, _ ->
+                                callingActivity.finish()
+                            }
                             .show()
                 }
             }
-        } else if (connectionMode.equals("OpenKeychain", ignoreCase = true)) {
+        } else if (connectionMode == ConnectionMode.OpenKeychain) {
             setAuthentication(username, identity).execute()
         } else {
-            val password = EditText(callingActivity)
-            password.hint = "Password"
-            password.width = LinearLayout.LayoutParams.MATCH_PARENT
-            password.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
-
-            MaterialAlertDialogBuilder(callingActivity)
-                    .setTitle(callingActivity.resources.getString(R.string.passphrase_dialog_title))
-                    .setMessage(callingActivity.resources.getString(R.string.password_dialog_text))
-                    .setView(password)
-                    .setPositiveButton(callingActivity.resources.getString(R.string.dialog_ok)) { _, _ ->
-                        // authenticate using the user/pwd and then execute the command
-                        setAuthentication(username, password.text.toString()).execute()
-                    }
-                    .setNegativeButton(callingActivity.resources.getString(R.string.dialog_cancel)) { _, _ ->
-                        callingActivity.finish()
+            @SuppressLint("InflateParams") val dialogView = callingActivity.layoutInflater.inflate(R.layout.git_passphrase_layout, null)
+            val passwordView = dialogView.findViewById<TextInputEditText>(R.id.git_auth_passphrase)
+            val password = encryptedSettings.getString("https_password", null)
+            if (password != null && password.isNotEmpty()) {
+                setAuthentication(username, password).execute()
+            } else {
+                val dialog = MaterialAlertDialogBuilder(callingActivity)
+                        .setTitle(callingActivity.resources.getString(R.string.passphrase_dialog_title))
+                        .setMessage(callingActivity.resources.getString(R.string.password_dialog_text))
+                        .setView(dialogView)
+                        .setPositiveButton(callingActivity.resources.getString(R.string.dialog_ok)) { _, _ ->
+                            val rememberPassphrase = dialogView.findViewById<MaterialCheckBox>(R.id.git_auth_remember_passphrase).isChecked
+                            if (rememberPassphrase) {
+                                encryptedSettings.edit().putString("https_password", passwordView.text.toString()).apply()
```

Comment:
The remembered passphrase should be deleted whenever the hostname changes to prevent accidental password leaks.

Reply:
d1151a0c2f4c